ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -523,6 +524,29 @@ class TreegraphSeries extends TreemapSeries {
 
             // Render link labels.
             seriesProto.drawDataLabels.call(this, this.links);
+
+            if (!this.options.dataLabels[0].allowOverlap) {
+                const getLabels = (
+                    labelOwners: TreegraphLink[] | TreegraphPoint[]
+                ): SVGElement[] => {
+                    const labels = [];
+
+                    for (let i = labelOwners.length - 1; i; i--) {
+                        const dataLabel = labelOwners[i]?.dataLabel;
+
+                        if (dataLabel) {
+                            labels.push(dataLabel);
+                        }
+                    }
+
+                    return labels;
+                };
+
+                this.chart.hideOverlappingLabels([
+                    ...getLabels(this.points),
+                    ...getLabels(this.links)
+                ]);
+            }
```

Comment:
I am not sure if that's a perfect solution for that issue. I think it would be much cleaner if we add only one line here: https://github.com/highcharts/highcharts/blob/ce0d44730018c8b6a9eceb0b081ba1c43f05419b/ts/Extensions/OverlappingDataLabels.ts#L329
something like `push(series.links || []);`.

Reply:
Ah, I can try to rewrite it to something like that!

Reply:
Its not exactly like that now, but it is way way simpler as it turns out only the link-labels was needed!